src/translators/crate.py

```diff
@@ -234,14 +234,18 @@ def _insert_entities_of_type(self,
                 original_attrs[col] = (attr, attr_t)
 
                 if attr_t not in NGSI_TO_CRATE:
-                    supported_types = ', '.join(NGSI_TO_CRATE.keys())
-                    msg = ("'{}' is not a supported NGSI type. "
-                           "Please use any of the following: {}. "
-                           "Falling back to {}.")
-                    self.logger.warning(msg.format(
-                        attr_t, supported_types, NGSI_TEXT))
-
-                    table[col] = NGSI_TO_CRATE[NGSI_TEXT]
+                    # if attribute is complex assume it as 
+                    if attr:
```

Comment:
you should check that the value is an object, it would say

Reply:
Yes, this will make everything fall here because 'attr' is the name of the attribute, always non-empty.

I suggest this approach:

1) develop a new function that receives a python object and by python introspection returns the corresponding NGSI type constant. Add a unit-tests for it in `test_crate.py`. E.g
```
assert translator.get_ngsi_type(1) == translator.NGSI_NUMBER
assert translator.get_ngsi_type("hello") == translator.NGSI_TEXT
assert translator.get_ngsi_type({'zip': 5000, 'country': 'FOO'}) == translator.NGSI_STRUCTURED_VALUE
... (and so on for all NGSI types)
```

2) Use this function within this `if attr_t not in NGSI_TO_CRATE:` block. You'll pass the attribute's value (typically `e[attr]['value']` but it could be `e[attr]` if the entity comes in key-value pairs. (see line 227). Key-value pairs is not yet fully supported, but just to be cautious for when it will.
You may still want to use the log, just instead of falling back to `NGSI_TEXT`, you say your best-effort guess determined to be NGSI_X (the return of your function). At least during dev this will be useful. Then translate such NGSI_X and that's the crate type you'll set to table[col] like ex line 244 did.

3) The second test you'll want to add is very similar to https://github.com/smartsdk/ngsi-timeseries-api/blob/7b3902df42aaffec65cbdc2498ddf97e31128228/src/translators/tests/test_crate.py#L183, but your 'foo' will be of `PostalAddress` type and its value a complete dict, like the one your breaking use case has.

4) Make sure all translator tests run locally. I may need to improve 'contributors' documentation on how to run these, ping me when you reach this point. When translator tests pass, then check all tests (from the root of the project).

The "metadata" part mentioned in the issue may be already taken care of for you by line 234. We'll see depending on your results of step 3.

Reply:
OK got it, I think I can go forward with your detailed advice
Thanks a lot for your reactivity and helpful hints.
I'll come to you in case of need.

Reply:
think I have managed to fix the the acceptation of unknown ngsi structured type:

function `_attr_is_structured` (issue #129)
I have added a test function: `test_accept_unknown_ngsi_type`, but not tested it locally

In addition I have fixed the issue #128  "-" in attributes and type not accepted by crate (quoting of tables and columns names, fix the problem)
I assume it's a good practice to add a test fonction then I added: `test_accept_special_chars`

I have tested on my own configuration with Orion latest, and crate 3.0.5, and it works fine 

I ran `sh run_tests.sh` from repository root directory....
Think I missed something to setup the test environment...
Time to ping you for a bit of help on this.